src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28567,6 +28820,53 @@ public class C
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void DisallowNull_Property_InDeconstructionAssignment()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [DisallowNull]public int? P1 { get; set; } = null;
+    void M()
+    {
+        (P1, _) = (null, 1); // 1
```

Comment:
have we tested that assigning non-null when `DisallowNull` is present updates the flow state appropriately?

Reply:
yes, here's an example: `DisallowNull_Field_NullInitializer`

---
In reply to: [324902150](https://github.com/dotnet/roslyn/pull/38610#discussion_r324902150) [](ancestors = 324902150)